src/AwsMetadata.js

```diff
@@ -53,6 +53,7 @@ export default class AwsMetadata {
       this.lookupMetadataKey(`network/interfaces/macs/${results.mac}/vpc-id`, (err, vpcId) => {
         results['vpc-id'] = vpcId;
         this.logger.debug('Found Instance AWS Metadata', results);
+        Object.keys(results).filter(k => results[k] === null).forEach(k => delete results[k]);
```

Comment:
Thanks for updating the PR. I think @kesne touched on this briefly in his earlier comments but should this be a ```==``` comparison instead of ```===``` so that we're checking for undefined?